src/portable/ehci/ehci.c

```diff
@@ -388,6 +390,8 @@ bool hcd_edpt_open(uint8_t rhport, uint8_t dev_addr, tusb_desc_endpoint_t const
 
   // TODO might need to disable async/period list
   list_insert(list_head, (ehci_link_t*) p_qhd, EHCI_QTYPE_QHD);
+  // Flush data cache if required, see hcd.h for details
+  hcd_dcache_flush((uint32_t)p_qhd, sizeof(ehci_qhd_t));
```

Comment:
for all calls of weak functions, it must be check with `if` first. Since not all compiler could skip un-implemented weak function (which is typically resolved to NULL in symbol table).

```C
if (hcd_dcache_flush) 
{
  hcd_dcache_flush((uint32_t)ep_desc, sizeof(tusb_desc_endpoint_t));
}
```

Reply:
I wasn't aware of this, so no problem I will make the changes. 🙂